src/js/server/utils.js

```diff
@@ -0,0 +1,140 @@
+/*
+
+    Utilities methods for performing user operations. These utilities should 
+    be usable from both http endpoints, and any other locations that can
+    consume promise based API's. 
+
+    Utilities are included for:
+    - Creating new user accounts
+    - Unlocking user accounts with a username/email and password.
+
+ */
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var request = require("request"); // TODO:  Replace this with a writable data source.
+
+var gpii = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.express.user.utils");
+
+fluid.defaults("gpii.express.user.utils", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        createNewUser: {
+            funcName: "gpii.express.user.utils.createNewUser",
+            args: ["{that}", "{arguments}.0"] // options
+        },
+        unlockUser: {
+            funcName: "gpii.express.user.utils.unlockUser",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"] // username, password
+        }
+    },
+    byUsernameOrEmailUrl:    {
+        expander: {
+            funcName: "fluid.stringTemplate",
+            args:     [ "%userDbUrl/_design/lookup/_view/byUsernameOrEmail?key=\"%username\"", "{that}.options.couch"]
+        }
+    },
+    writeUrl: "{that}.options.couch.userDbUrl",
+    saltLength:       32,
+    verifyCodeLength: 16,
+    codeKey:          "verification_code",  // Must match the value in gpii.express.user.verify
+    rules: {
+        createUserWrite: {
+            "name":          "username", // Default rules are designed to cater to CouchDB  and express-couchUser conventions, but can be overriden.
+            "username":      "username",
+            "email":         "email",
+            roles:           { literalValue: []},
+            type:            { literalValue: "user"},
+            password_scheme: { literalValue: "pbkdf2"},
+            iterations:      { literalValue: 10},
+            verified:        { literalValue: false}
+        }
+    },
+    components: {
+        byUsernameOrEmailReader: {
+            // TODO:  Replace with the new "asymmetric" dataSource once that code has been reviewed
+            type: "gpii.express.user.couchdb.read",
+            options: {
+                url: "{gpii.express.user.utils}.options.byUsernameOrEmailUrl",
+                rules: {
+                    read: {
+                        "":         "rows.0.value",
+                        "username": "rows.0.value.name"
+                    }
+                },
+                termMap: { username: "%username"}
+            }
+        }
+    }
+});
+
+gpii.express.user.utils.createNewUser = function (that, userData) {
```

Comment:
If devs are now expected to be able to call these functions directly without using the API, it seems like it's time to add simple JSDocs.

Reply:
Should be "user's" throughout.